pkg/bootstrapper/bootstrapper.go

```diff
@@ -96,14 +103,14 @@ const (
 // These regex are global, so that we only need to compile them once
 var (
 	// cloudProviderRegex searches for the cloud provider option given to the kubelet
-	cloudProviderRegex = regexp.MustCompile(`--cloud-provider=(\w*)`)
+	cloudProviderRegex = regexp.MustCompile(`--` + cloudProviderOption + `=(.*)\n`)
```

Comment:
Because the cause of the bug is that any of the kubelet options can be set to an environment variable. Fixing just the verbosity argument does not fix this bug. It fixes a symptom of the bug but not the cause.

Reply:
The bug is only with the verbosity argument at this point so you are (or should be) only fixing the bug and not a symptom.